src/Components/Web/src/Forms/InputFile/BrowserFileStream.cs

```diff
@@ -53,7 +53,10 @@ public override Task<int> ReadAsync(byte[] buffer, int offset, int count, Cancel
 
         public override async ValueTask<int> ReadAsync(Memory<byte> buffer, CancellationToken cancellationToken = default)
         {
-            int maxBytesToRead = (int)(Length - Position);
+            var bytesAvailableToRead = Length - Position;
+            var maxBytesToRead = bytesAvailableToRead > int.MaxValue ?
+                int.MaxValue :
+                (int)bytesAvailableToRead;
```

Comment:
What's this new check doing?

Reply:
This is protecting against int overflow. 

Ex. Consider `Int32.MaxValue = 2147483647` ~= 2.14GB. So if you have a 3GB file, then we're trying to cast a number larger than `Int32.MaxValue` into an int, thus leading to a negative value. This would subsequently fail on line 66 `maxBytesToRead <= 0` and `return 0`.

Reply:
It's fine if you want to leave it as-is (though maybe consider adding a comment, as it's a bit unusual).

However is it possible there might be a more basic solution? Something like:

```cs
var maxBytesToRead = (int)Math.Min(bytesAvailableToRead, (long)buffer.Length);
```

... and also removing lines 61-64.

Of course it's possible I'm misreading the logic here and this doesn't account for some case :)